constraints/src/main/java/gov/nasa/jpl/aerie/constraints/time/IntervalMap.java

```diff
@@ -196,6 +197,27 @@ public <R> IntervalMap<R> map(final BiFunction<V, Interval, R> transform) {
     return builder.build();
   }
 
+  /**
+   * A similar operation to {@link IntervalMap#map(Function)}, except it allows producing any number of segments
+   * per interval. Each segment is then flattened into a new IntervalMap.
+   */
+  public <R> IntervalMap<R> flatMap(final Function<V, Stream<Segment<R>>> transform) {
```

Comment:
Just musing - it is important that the segments returned by `transform` be contained in the original overlapping interval?

Reply:
As long as the intervalmap user is OK with last-one-wins behavior, no.